mjcf_to_sdformat/mjcf_to_sdformat/converters/geometry.py

```diff
@@ -0,0 +1,130 @@
+
+"""Module to convert MJCF geoms to SDFormat Collision/Visual geometries"""
+
+from ignition.math import Vector2d, Vector3d
+
+import sdformat as sdf
+import sdformat_mjcf_utils.sdf_utils as su
+
+VISUAL_NUMBER = 0
+COLLISION_NUMBER = 0
+
+
+def add_mjcf_geometry_to_sdf(geom):
+    """
+    Converts an MJCF geom to a SDFormat geometry.
+
+    :param mjcf.Element geom: The MJCF geom
+    :return: The newly created SDFormat geometry.
+    :rtype: sdf.Geometry
+    """
+    sdf_geometry = sdf.Geometry()
+    if geom.type == "box":
+        box = sdf.Box()
+        box.set_size(su.list_to_vec3d(geom.size) * 2)
+        sdf_geometry.set_box_shape(box)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.BOX)
+    elif geom.type == "capsule":
+        capsule = sdf.Capsule()
+        capsule.set_radius(geom.size[0])
+        if geom.fromto is None:
+            capsule.set_length(geom.size[1] * 2)
+        else:
+            v1 = Vector3d(geom.fromto[0], geom.fromto[1], geom.fromto[2])
+            v2 = Vector3d(geom.fromto[3], geom.fromto[4], geom.fromto[5])
+            length = v1.distance(v2)
+            capsule.set_length(length)
+        sdf_geometry.set_capsule_shape(capsule)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.CAPSULE)
+    elif geom.type == "cylinder":
+        cylinder = sdf.Cylinder()
+        cylinder.set_radius(geom.size[0])
+        if geom.fromto is None:
+            cylinder.set_length(geom.size[1] * 2)
+        else:
+            v1 = Vector3d(geom.fromto[0], geom.fromto[1], geom.fromto[2])
+            v2 = Vector3d(geom.fromto[3], geom.fromto[4], geom.fromto[5])
+            length = v1.distance(v2)
+            cylinder.set_length(length)
+        sdf_geometry.set_cylinder_shape(cylinder)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.CYLINDER)
+    elif geom.type == "ellipsoid":
+        ellipsoid = sdf.Ellipsoid()
+        ellipsoid.set_radii(su.list_to_vec3d(geom.size))
+        sdf_geometry.set_ellipsoid_shape(ellipsoid)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.ELLIPSOID)
+    elif geom.type == "sphere":
+        sphere = sdf.Sphere()
+        sphere.set_radius(geom.size[0])
+        sdf_geometry.set_sphere_shape(sphere)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.SPHERE)
+    elif geom.type == "plane":
+        plane = sdf.Plane()
+        plane.set_size(Vector2d(geom.size[0] * 2, geom.size[1] * 2))
+        sdf_geometry.set_plane_shape(plane)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.PLANE)
+    else:
+        raise RuntimeError(
+            f"Encountered unsupported shape type {geom.type}")
+    return sdf_geometry
+
+
+def add_mjcf_visual_to_sdf(geom):
+    """
+    Converts MJCF geom to a SDFormat visual
+    MJCF geom should be part of group `VISUAL_GEOM_GROUP`.
+
+    :param mjcf.Element geom: The MJCF geom.
+    :return: The newly created SDFormat visual.
+    :rtype: sdformat.Visual
+    """
+    visual = sdf.Visual()
```

Comment:
Should the pose of the visual be set here from the geom?

Reply:
https://github.com/gazebosim/gz-mujoco/blob/40148462713bdbba5fa3ba236d57bf8c24596e61/mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py#L99-L128